tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/TypeUtils.java

```diff
@@ -94,6 +94,8 @@ public final class TypeUtils {
 		PRIMITIVES.put( TypeKind.DOUBLE, "double" );
 	}
 
+	private static final Map<Element, GenericTypeParameterResolver> typeResolversMap = new HashMap<>();
+
```

Comment:
I don't think we can cache stuff like this in a `static`. As I understand it, we have no control over the lifecycle of these classes, and they could in principle fill up with all kinds of stuff over many calls to the processor.

Reply:
I will remove class `GenericTypeParameterResolver` and move resolving into `TypeUtils` for each call.